src/main/java/org/jboss/pnc/bifrost/source/db/FinalLog.java

```diff
@@ -96,4 +104,25 @@ public static Collection<FinalLog> getFinalLogsWithoutPreviousRetries(long proce
         }
         return logMap.values();
     }
+
+    public static long deleteByProcessContext(long processContext, String tag, boolean temporaryOnly) {
+        // language=HQL
+        String logEntryQuery = "select id from LogEntry where processContext = :processContext";
+        Parameters parameters = Parameters.with("processContext", processContext);
+        if (temporaryOnly) {
+            logEntryQuery += " and temporary = :temporary";
+            parameters.and("temporary", true);
+        }
+
+        // unfortunately JPA/HQL DELETE queries do not support Joins in FROM clause, so we have fallback to subquery
+        // in WHERE clause
+        // language=HQL
```

Comment:
sot his wouldn't work?
```
from FinalLog l where l.logEntry.processContext = :processContext and l.logEntry.temporary = :temporary and :tag in elements(l.tags)
```

Reply:
no unfortunately, I've tried a lot of variants like this